crates/bdk/src/wallet/mod.rs

```diff
@@ -1006,7 +1006,7 @@ impl<D> Wallet<D> {
     /// # let mut wallet: Wallet<()> = todo!();
     /// # let txid:Txid = todo!();
     /// let tx = wallet.get_tx(txid).expect("transaction").tx_node.tx;
-    /// let (sent, received) = wallet.sent_and_received(tx);
+    /// let (sent, received) = wallet.sent_and_received(&tx);
```

Comment:
Interested to understand why this `&` is needed now. Before tx was of type `Transaction` right, so shouldn't it needed to have been borrowed before?

Reply:
Before, `TxNode`'s `tx` field is of a reference to `T`. I changed it in this PR to just be `tx: T`. Note that `T` is now `Arc<Transaction>`.

Because `sent_and_received` takes in a `&Transaction`, we expect to call `.as_ref` on `Arc<Transaction>`. However, we can make use of the [auto-deref properties of `AsRef` implementations](https://doc.rust-lang.org/std/convert/trait.AsRef.html#generic-implementations) (such as `Arc`), so just doing `&tx` is enough.

Reply:
Maybe we can map to shared ref `&` internally and avoid returning `Arc<Transaction>` in `get_tx{_node}`

Reply:
Yes, I think we should return a concrete `Arc` for this API to be useful (cheaply share transactions between chain-source, receiving-structs and changesets).